src/Rector/Deprecation/DrupalServiceRenameBase.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace DrupalRector\Rector\Deprecation;
+
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+
+/**
+ * Renames deprecated Drupal::services().
+ *
+ * What is covered:
+ *  - Renames argument in Drupal::service() calls.
+ */
+abstract class DrupalServiceRenameBase extends StaticArgumentRenameBase {
```

Comment:
Since this seems like a common use case that you pointed out, I created this base class. This base class is built off the `StaticArgumentRenameBase` which could have broader use for other methods and classes. The two have the split pieces of the code you sent.

Reply:
The bases classes are great.  It took me a while to realize they existed; they get lost in between the non-base classes.  I didn't realize until now that class names ending with *Base.php were all the  re-usable utility classes and that I could find them by looking at the class names.  D'oh!  One developer experience (DX) improvement could be to put them in their own sub-package.

Reply:
Thanks for the feedback!

I was finding that a bit as well. We're getting a lot more files now. I'll discuss this with our internal team.